submission-panels.md

```diff
@@ -0,0 +1,80 @@
+# Submission-panels Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**/api/configuration/submission-definitions**   
+
+Provide access to the configured submission definitions. It returns the list of existent submission-definitions.
+
+Example: to be provided
+
+## Single Submission-panels 
+**/api/configuration/submission-definitions/<:definition-name>**
+
+Provide detailed information about a specific submission-definition. The JSON response document is as follow
+```json
+{
+  "name": "traditional",
+  "panels": [
+  {
+  	id: "id-of-the-input-form-page",
+  	header: "First page",
+  	mandatory: true,
+  	type: "inputform",
+  	scope: null,
+  	_links: {
+  		"configuration" : "<dspace-url>/configuration/inputforms/<:id-of-the-input-form-page>" 
```

Comment:
These `panel` definitions are so "light" on content, that a part of me does wonder why we don't consider embedding the `fields` into this response (and combine the two endpoints together). 

Another option here might be to organize these endpoints more hierarchically:
* `/api/config/submission-forms` -> List of all forms (i.e. submission-definitions) by name
* `/api/config/submission-forms/<:name>`  -> Higher level info about one form
* `/api/config/submission-forms/<:name>/pages` -> List of pages (i.e. panel) in this form
* `/api/config/submission-forms/<:name>/pages/<:id>` -> Detail of a specific page/panel

To me at least, it seems like we should be able to have this all under one hierarchical endpoint, even if underneath it's pulling info from separate config files.

Reply:
They are not hierarchical because the same panel can be reused in other definitions. This is already the case in the current configuration and it is a very convenient way to manage large definitions. Please note that in the last commits we have introduced a /api/config/submission-panels/<:panel-id> endpoint.
https://github.com/DSpace/Rest7Contract/pull/11/commits/3fd9c24e91de383b514de0646e3f522c562e5284
I strongly believe that the configuration need to go on a separate, specialized endpoint as it depends on the type of panel that you need to instantiate. I don't want to limit the panel, that are the primary way to extend the DSpace submission, to be necessary something form based.

Reply:
@abollini : I see your point. The forms are meant to be reusable, and should therefore *not* be only accessible off the parent endpoint. So, the idea of a hierarchy doesn't work well here.

I like the idea of moving this then to two endpoints that are prefixed similarly (to show they are related).  So, something like:

* `/api/config/submission-definitions`  = Endpoint for `item-submission.xml` (links together forms into a process)
* `/api/config/submission-forms` = Endpoint for `input-forms.xml` (defines reusable forms which are just sets of fields)